pkg/manifests/manifests_test.go

```diff
@@ -1046,6 +1052,92 @@ ingress:
 	}
 }
 
+func TestAdditionalAlertManagerConfigsSecret(t *testing.T) {
+	c, err := NewConfigFromString(`prometheusK8s:
+  additionalAlertManagerConfigs:
+  - apiVersion: v2
+    bearerToken:
+      name: alertmanager1-bearer-token
+      key: token
+    scheme: https
+    tlsConfig:
+      ca: 
+        name: alertmanager1-ca
+        key: root-ca.crt
+    pathPrefix: /
+    staticConfigs:
+    - alertmanager1-remote.com
+    - alertmanager1-remotex.com
+  - apiVersion: v2
+    scheme: https
+    timeout: 60s
+    tlsConfig:
+      ca: 
+        name: alertmanager2-cert
+        key: root-ca.crt
+      cert:
+        name: alertmanager2-cert
+        key: cert.crt
+      key:
+        name: alertmanager2-cert
+        key: key.crt
+    pathPrefix: /
+    staticConfigs:
+    - alertmanager2-remote.com
+`)
+	expected := `- scheme: https
+  path_prefix: /
+  api_version: v2
+  authorization:
+    credentials_file: /etc/prometheus/secrets/alertmanager1-bearer-token/token
+  tls_config:
+    ca_file: /etc/prometheus/secrets/alertmanager1-ca/root-ca.crt
+    insecure_skip_verify: false
+  static_configs:
+  - targets:
+    - alertmanager1-remote.com
+    - alertmanager1-remotex.com
+- scheme: https
+  path_prefix: /
+  api_version: v2
+  timeout: 60s
+  tls_config:
+    ca_file: /etc/prometheus/secrets/alertmanager2-cert/root-ca.crt
+    cert_file: /etc/prometheus/secrets/alertmanager2-cert/cert.crt
+    key_file: /etc/prometheus/secrets/alertmanager2-cert/key.crt
+    insecure_skip_verify: false
+  static_configs:
+  - targets:
+    - alertmanager2-remote.com
+`
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", c, defaultInfrastructureReader(), &fakeProxyReader{}, NewAssets(assetsPath))
+
+	p, err := f.PrometheusK8s(
+		"prometheus-k8s.openshift-monitoring.svc",
+		&v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+		&v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+	)
+
+	if !strings.Contains(strings.Join(p.Spec.Secrets, ","), "alertmanager2-cert") ||
+		!strings.Contains(strings.Join(p.Spec.Secrets, ","), "alertmanager1-ca") ||
+		!strings.Contains(strings.Join(p.Spec.Secrets, ","), "alertmanager1-bearer-token") {
+		t.Fatal("Prometheus secrets are not generated correctly, expected to have alertmanager2-cert/alertmanager1-ca/alertmanager1-bearer-token secret, but found", p.Spec.Secrets)
+	}
```

Comment:
Could you iterate over `p.Spec.Secrets` and check that all expected secrets are present?

```suggestion
	secrets := make(map[string]struct{})
	for _, s := range p.Spec.Secrets {
		secrets[s] = struct{}{}
	}

	for _, exp := range []string{"alertmanager2-cert",  "alertmanager1-ca", "alertmanager1-bearer-token"} {
		if _, found := secrets[exp]; found {
			continue
		}
		t.Fatal("Prometheus secrets are not generated correctly, expected to have %s but got none", exp)
	}
```